src/Tvheadend.cpp

```diff
@@ -478,6 +482,11 @@ PVR_ERROR CTvheadend::GetRecordings ( ADDON_HANDLE handle )
         }
       }
 
+      /* Fill recording url */
+      if (_http) {
+        snprintf(rec.strStreamURL, sizeof(rec.strStreamURL), "http://%s:%d/dvrfile/%s",host.c_str(), port, rec.strRecordingId);
```

Comment:
As I said before, I'm afraid that using PVR_RECORDING.strStreamUrl will not work any longer with Kodi v18 Leia. Thus, this approach could be a dead end. @FernetMenta please comment.